tests/smoke/pool_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package smoke
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/rook/rook/pkg/model"
+	"github.com/stretchr/testify/require"
+)
+
+// Smoke Test for Mon failover - Test check the following operations for the Mon failover in order
+//Delete mon pod, Wait for new mon pod
+func (suite *SmokeSuite) TestPoolResize() {
+	logger.Infof("Pool Resize Smoke Test")
+
+	poolName := "testpool"
+
+	pClient := suite.helper.GetPoolClient()
+
+	pool := model.Pool{
+		Name: poolName,
+		ReplicationConfig: model.ReplicatedPoolConfig{
+			Size: 1,
+		},
+	}
+
+	out, err := pClient.PoolCreate(pool)
+	logger.Infof("poolCreate: %+v", out)
+	require.Nil(suite.T(), err)
+
+	poolFound := false
+
+	// Wait for pool to appear
+	for i := 0; i < 10; i++ {
+		pools, err := pClient.PoolList()
+		require.Nil(suite.T(), err)
+		for _, p := range pools {
+			if p.Name != poolName {
+				logger.Infof("created pool %s not found in pool list", poolName)
+				continue
+			}
+			poolFound = true
+		}
+		if poolFound {
+			break
+		}
+
+		logger.Infof("Waiting for pool to appear")
+		time.Sleep(2 * time.Second)
+	}
+
+	require.Equal(suite.T(), true, poolFound, "pool not found")
+
+	pool.ReplicationConfig.Size = 3
+
+	out, err = pClient.PoolCreate(pool)
+	logger.Infof("poolCreate (modify): %+v", out)
+	require.Nil(suite.T(), err)
+
+	poolFound = false
+	// Wait for pool resize to happen
+	for i := 0; i < 10; i++ {
+		pools, err := pClient.PoolList()
+		require.Nil(suite.T(), err)
+		for _, p := range pools {
+			if p.Name != poolName {
+				logger.Infof("created pool %s not found in pool list", poolName)
```

Comment:
this message is a bit verbose since there are multiple other pools in the cluster created by other tests

Reply:
You're right. I'll remove it.
Are these messages at https://github.com/rook/rook/pull/991/files/812797606fef84a40b5c27cf7d82ad2fc84e97ed#diff-b72b3bcbe5a6b22165802a5495d51703R98 and https://github.com/rook/rook/pull/991/files/812797606fef84a40b5c27cf7d82ad2fc84e97ed#diff-b72b3bcbe5a6b22165802a5495d51703R64 "enough"?